app/components/mobilePaperShow/mobilePaperShow.tsx

```diff
@@ -51,94 +51,90 @@ const MobilePaperShow: React.FC<MobilePaperShowProps> = ({ paper, location }) =>
-                currentScrollTop < citedSectionOffsetTop &&
-                lastPosition.current !== 'onRefList'
-              ) {
-                setCurrentPosition('onRefList');
-                lastPosition.current = 'onRefList';
-              } else if (currentScrollTop >= citedSectionOffsetTop && lastPosition.current !== 'onCitedList') {
-                setCurrentPosition('onCitedList');
-                lastPosition.current = 'onCitedList';
-              }
+            if (currentScrollTop < buttonGroupOffsetTop && lastPosition.current !== 'abovePaperInfo') {
+              setCurrentPosition('abovePaperInfo');
+              lastPosition.current = 'abovePaperInfo';
+            } else if (
+              currentScrollTop >= buttonGroupOffsetTop &&
+              currentScrollTop < relatedTabWrapperOffsetTop &&
+              lastPosition.current !== 'underPaperInfo'
+            ) {
+              setCurrentPosition('underPaperInfo');
+              lastPosition.current = 'underPaperInfo';
+            } else if (
+              currentScrollTop >= relatedTabWrapperOffsetTop &&
+              currentScrollTop < refSectionOffsetTop &&
+              lastPosition.current !== 'onRelatedList'
+            ) {
+              setCurrentPosition('onRelatedList');
+              lastPosition.current = 'onRelatedList';
+            } else if (
+              currentScrollTop >= refSectionOffsetTop &&
+              currentScrollTop < citedSectionOffsetTop &&
+              lastPosition.current !== 'onRefList'
+            ) {
+              setCurrentPosition('onRefList');
+              lastPosition.current = 'onRefList';
+            } else if (currentScrollTop >= citedSectionOffsetTop && lastPosition.current !== 'onCitedList') {
+              setCurrentPosition('onCitedList');
+              lastPosition.current = 'onCitedList';
+            }
+          } else {
+            if (currentScrollTop < buttonGroupOffsetTop && lastPosition.current !== 'abovePaperInfo') {
+              setCurrentPosition('abovePaperInfo');
+              lastPosition.current = 'abovePaperInfo';
+            } else if (
+              currentScrollTop >= buttonGroupOffsetTop &&
+              currentScrollTop < refSectionOffsetTop &&
+              lastPosition.current !== 'underPaperInfo'
+            ) {
+              setCurrentPosition('underPaperInfo');
+              lastPosition.current = 'underPaperInfo';
+            } else if (
+              currentScrollTop >= refSectionOffsetTop &&
+              currentScrollTop < citedSectionOffsetTop &&
+              lastPosition.current !== 'onRefList'
+            ) {
+              setCurrentPosition('onRefList');
+              lastPosition.current = 'onRefList';
+            } else if (currentScrollTop >= citedSectionOffsetTop && lastPosition.current !== 'onCitedList') {
+              setCurrentPosition('onCitedList');
+              lastPosition.current = 'onCitedList';
```

Comment:
These huge lines of code are not neat...
It looks like there are some possible improvements available, like make this scroll handler as `hooks`.
How do you think about it?

Reply:
of course, it is possible. however, I don't think that is not proper to be in this commit.
and I think that's not a proper solution for detecting the position of the current screen.
let's improve it other commit.

Reply:
Gotit 🤟